src/echoclient/EchoClient.php

```diff
@@ -62,51 +66,51 @@ function __construct()
     {
         $class = ECHO_CLASS_PREFIX . $class;
-        $baseUrl = $this->getBaseUrl();
+        $eventJson = $this->getEventJson($class, $source, $props, $userId, $timestamp);
 
-        if (!$baseUrl) {
-            // fail silently when creating events, should not stop user interaction as echo events are collected on a best-endeavours basis
-            $this->getLogger()->warning('Echo server is not defined (missing ECHO_HOST define), not sending event - ' . $class,
-                $props);
-            return false;
+        return $this->sendJsonEventDataToEcho($eventJson);
+    }
+
+    /**
+     * Adds multiple events to echo
+     *
+     * @param EchoEvent[] $events An array of EchoEvent objects
+     * @return bool True if successful, else false
+     */
+    public function createEvents(array $events)
+    {
+        if (count($events) > self::ECHO_MAX_BATCH_EVENTS) {
+            $this->getLogger()->warning('Batch contains more than ' . self::ECHO_MAX_BATCH_EVENTS . ' events');
+            throw new \Exception("Batch of events exceeds the maximum allowed size");
         }
 
-        $eventUrl = $baseUrl . '/events';
-        $eventJson = $this->getEventJson($class, $source, $props, $userId);
+        foreach ($events as $event) {
+            if (!$event instanceof \echoclient\EchoEvent) {
+                throw new \Exception("Batch must only contain EchoEvent objects");
+            }
+        }
 
-        try {
-            $client = $this->getHttpClient();
-            $request = $client->post($eventUrl, $this->getHeaders(), $eventJson, array('connect_timeout' => 2));
-            $response = $request->send();
+        $eventsJson = json_encode($events, true);
 
-            if ($response->isSuccessful()) {
-                $this->getLogger()->debug('Success sending event to echo - ' . $class, $props);
-                return true;
-            } else {
-                $this->getLogger()->warning('Failed sending event to echo - ' . $class, array(
-                    'responseCode' => $response->getStatusCode(),
-                    'responseBody' => $response->getBody(true),
-                    'requestProperties' => $props
-                ));
-                return false;
-            }
-        } catch (\Exception $e) {
-            // For any exception issue, just log the issue and fail silently.  E.g. failure to connect to echo server, or whatever.
-            $this->getLogger()->warning('Failed sending event to echo - ' . $class,
-                array('exception' => get_class($e), 'message' => $e->getMessage(), 'requestProperties' => $props));
-            return false;
+        // strlen returns no. bytes in a string.
+        $sizeOfBatch = strlen(utf8_decode($eventsJson));
+        if ($sizeOfBatch > self::ECHO_MAX_BATCH_SIZE_IN_BYTES) {
+            throw new \Exception("Batch must be less than 1mb in size");
```

Comment:
Different custom exception subtype - perhaps they should all be subtypes of [InvalidArgumentException](http://php.net/manual/en/class.invalidargumentexception.php)?